video/clip.go

```diff
@@ -0,0 +1,90 @@
+package video
+
+import (
+	"fmt"
+	"github.com/grafov/m3u8"
+	"github.com/livepeer/catalyst-api/log"
+)
+
+func getTotalDurationAndSegments(manifest *m3u8.MediaPlaylist) (float64, uint64) {
+	if manifest == nil {
+		return 0.0, 0
+	}
+	var totalDuration float64
+	var segmentCount uint64
+	for _, segment := range manifest.Segments {
+		if segment == nil {
+			break
+		}
```

Comment:
nit: just use `manifest.GetAllSegments()` then you can avoid this if statement and just take the length of that slice for segmentCount rather than needed to calculate yourself.

Reply:
Ah good point, I seem to have missed that method already exists in the m3u8 library. Will update.

Reply:
Updated